[8.17] [Lens] fix drag and drop ghost after Chrome update to 134 (#213928)

kibanamachine opened the pull request:
# Backport

This will backport the following commits from `main` to `8.17`:
 - https://github.com/elastic/kibana/pull/213928
 - https://github.com/elastic/kibana/pull/215439

<!--- Backport version: 9.6.6 -->

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sorenlouv/backport)

<!--BACKPORT [{"author":{"name":"Marta Bondyra","email":"4283304+mbondyra@users.noreply.github.com"},"sourceCommit":{"committedDate":"2025-03-12T11:30:51Z","message":"[Lens] fix drag and drop ghost after Chrome update to 134 (#213928)\n\n## Summary\nFixes https://github.com/elastic/kibana/issues/213708\n\nFixes the ghost display. As @nickofthyme pointed out\n[here](https://github.com/elastic/kibana/issues/213708#issuecomment-2712076212),\nthe reason was using inset-inline-start: -10000px for the keyboard\nhandler that is one of the children of the draggable component. The\nquick fix is to just move it outside of the html element that is\ndragged. @markov00 please assesswhere it should be merged 🙏🏼\n\n---------\n\nCo-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>","sha":"84ae9c98c208957b3efc02c6b00bed8d567d4209","branchLabelMapping":{"^v9.1.0$":"main","^v8.19.0$":"8.x","^v(\\d+).(\\d+).\\d+$":"$1.$2"}},"sourcePullRequest":{"labels":["Feature:Discover","release_note:fix","Team:Visualizations","Feature:Lens","backport:prev-minor","backport:prev-major","Feature:UnifiedFieldList","v9.1.0"],"title":"[Lens] fix drag and drop ghost after Chrome update to 134","number":213928,"url":"https://github.com/elastic/kibana/pull/213928","mergeCommit":{"message":"[Lens] fix drag and drop ghost after Chrome update to 134 (#213928)\n\n## Summary\nFixes https://github.com/elastic/kibana/issues/213708\n\nFixes the ghost display. As @nickofthyme pointed out\n[here](https://github.com/elastic/kibana/issues/213708#issuecomment-2712076212),\nthe reason was using inset-inline-start: -10000px for the keyboard\nhandler that is one of the children of the draggable component. The\nquick fix is to just move it outside of the html element that is\ndragged. @markov00 please assesswhere it should be merged 🙏🏼\n\n---------\n\nCo-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>","sha":"84ae9c98c208957b3efc02c6b00bed8d567d4209"}},"sourceBranch":"main","suggestedTargetBranches":[],"targetPullRequestStates":[{"branch":"main","label":"v9.1.0","branchLabelMappingKey":"^v9.1.0$","isSourceBranch":true,"state":"MERGED","url":"https://github.com/elastic/kibana/pull/213928","number":213928,"mergeCommit":{"message":"[Lens] fix drag and drop ghost after Chrome update to 134 (#213928)\n\n## Summary\nFixes https://github.com/elastic/kibana/issues/213708\n\nFixes the ghost display. As @nickofthyme pointed out\n[here](https://github.com/elastic/kibana/issues/213708#issuecomment-2712076212),\nthe reason was using inset-inline-start: -10000px for the keyboard\nhandler that is one of the children of the draggable component. The\nquick fix is to just move it outside of the html element that is\ndragged. @markov00 please assesswhere it should be merged 🙏🏼\n\n---------\n\nCo-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>","sha":"84ae9c98c208957b3efc02c6b00bed8d567d4209"}}]}] BACKPORT-->

mbondyra committed:
[Lens] fix drag and drop ghost after Chrome update to 134 (#213928)

## Summary
Fixes https://github.com/elastic/kibana/issues/213708

Fixes the ghost display. As @nickofthyme pointed out
[here](https://github.com/elastic/kibana/issues/213708#issuecomment-2712076212),
the reason was using inset-inline-start: -10000px for the keyboard
handler that is one of the children of the draggable component. The
quick fix is to just move it outside of the html element that is
dragged. @markov00 please assesswhere it should be merged 🙏🏼

---------

Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>
(cherry picked from commit 84ae9c98c208957b3efc02c6b00bed8d567d4209)

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/283191)
* Commit: 98556375406b43f0c15bd3f0dc5d2a517ce45441

### Failed CI Steps
* [FTR Configs #55](https://buildkite.com/elastic/kibana-pull-request/builds/283191#01958a48-e6b7-4b05-935d-e20672ce53fd)
* [FTR Configs #55](https://buildkite.com/elastic/kibana-pull-request/builds/283191#01958a69-9196-47a0-b09e-08aff4f5d223)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/283191#01958a48-e6b7-4b05-935d-e20672ce53fd) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/283191/jobs/01958a48-e6b7-4b05-935d-e20672ce53fd/artifacts/01958a69-839c-4eff-83e4-cb554f028707) FTR Configs #<span></span>55 / lens app - group 5 lens drag and drop tests keyboard drag and drop should drop a field to workspace
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/283191#01958a69-9196-47a0-b09e-08aff4f5d223) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/283191/jobs/01958a69-9196-47a0-b09e-08aff4f5d223/artifacts/01958a76-e34c-4a72-b9a7-3bd8bb382c87) FTR Configs #<span></span>55 / lens app - group 5 lens drag and drop tests keyboard drag and drop should drop a field to workspace

cc @mbondyra

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/283191","number":283191,"commit":"98556375406b43f0c15bd3f0dc5d2a517ce45441"}],"number":283191}
buildkite-pr-comment-->

mbondyra committed:
[Lens] fix drag and drop failing test (#215439)

## Summary

Fixes https://github.com/elastic/kibana/issues/213324

For some reason this PR https://github.com/elastic/kibana/pull/213928
didn't catch that the changes make the test fail. This PR fixes it.
After this, we'll be able to backport the original PR.
(The pattern of using `data-attr-field` is already used in Discover so I
think it's a good opportunity to align)

---------

Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>
Co-authored-by: Nick Partridge <nicholas.partridge@elastic.co>

nickofthyme committed:
Merge branch '8.17' into backport/8.17/pr-213928
